pkg/cmd/server/kubernetes/master/master_config_test.go

```diff
@@ -31,51 +31,86 @@ func TestAPIServerDefaults(t *testing.T) {
 	// This is a snapshot of the default config
 	// If the default changes (new fields are added, or default values change), we want to know
 	// Once we've reacted to the changes appropriately in BuildKubernetesMasterConfig(), update this expected default to match the new upstream defaults
-	expectedDefaults := &apiserveroptions.ServerRunOptions{
-		GenericServerRunOptions: &genericapiserveroptions.ServerRunOptions{
-			AnonymousAuth:           false,
-			BindAddress:             net.ParseIP("0.0.0.0"),
-			CertDirectory:           "/var/run/kubernetes",
-			InsecureBindAddress:     net.ParseIP("127.0.0.1"),
-			InsecurePort:            8080,
-			LongRunningRequestRE:    "(/|^)((watch|proxy)(/|$)|(logs?|portforward|exec|attach)/?$)",
-			MaxRequestsInFlight:     400,
-			SecurePort:              6443,
-			EnableProfiling:         true,
-			EnableGarbageCollection: true,
-			EnableWatchCache:        true,
-			MinRequestTimeout:       1800,
-			ServiceNodePortRange:    genericapiserveroptions.DefaultServiceNodePortRange,
-			RuntimeConfig:           utilconfig.ConfigurationMap{},
-			StorageVersions:         kapi.Registry.AllPreferredGroupVersions(),
-			MasterCount:             1,
-			DefaultStorageVersions:  kapi.Registry.AllPreferredGroupVersions(),
+	expectedDefaults := &kubeapiserveroptions.ServerRunOptions{
+		ServiceNodePortRange: kubeapiserveroptions.DefaultServiceNodePortRange,
+		MasterCount:          1,
+		GenericServerRunOptions: &apiserveroptions.ServerRunOptions{
+			MaxRequestsInFlight:         400,
+			MaxMutatingRequestsInFlight: 200,
+			MinRequestTimeout:           1800,
+			AdmissionControl:            "AlwaysAdmit",
+		},
+		Etcd: &apiserveroptions.EtcdOptions{
 			StorageConfig: storagebackend.Config{
 				ServerList: nil,
 				Prefix:     "/registry",
 				DeserializationCacheSize: 0,
+				Copier: kapi.Scheme,
+			},
+			DefaultStorageMediaType: "application/vnd.kubernetes.protobuf",
+			DeleteCollectionWorkers: 1,
+			EnableGarbageCollection: true,
+			EnableWatchCache:        true,
+		},
+		SecureServing: &apiserveroptions.SecureServingOptions{
+			ServingOptions: apiserveroptions.ServingOptions{
+				BindAddress: net.ParseIP("0.0.0.0"),
+				BindPort:    6443,
 			},
-			DefaultStorageMediaType:                  "application/json",
-			AdmissionControl:                         "AlwaysAdmit",
-			AuthorizationMode:                        "AlwaysAllow",
-			DeleteCollectionWorkers:                  1,
-			MasterServiceNamespace:                   "default",
-			AuthorizationWebhookCacheAuthorizedTTL:   5 * time.Minute,
-			AuthorizationWebhookCacheUnauthorizedTTL: 30 * time.Second,
+			ServerCert: apiserveroptions.GeneratableKeyCert{
+				CertDirectory: "/var/run/kubernetes",
```

Comment:
do we need to unset this and PairName to ensure we don't ever try to generate this?